cmd/extproc/mainlib/main.go

```diff
@@ -21,59 +22,88 @@ import (
 	"github.com/envoyproxy/ai-gateway/internal/version"
 )
 
-// parseAndValidateFlags parses and validates the flags passed to the external processor.
-func parseAndValidateFlags(args []string) (configPath, addr string, logLevel slog.Level, err error) {
-	fs := flag.NewFlagSet("AI Gateway External Processor", flag.ContinueOnError)
-	configPathPtr := fs.String(
+// extProcFlags is the struct that holds the flags passed to the external processor.
+type extProcFlags struct {
+	configPath  string     // path to the configuration file
+	extProcAddr string     // gRPC address for the external processor
+	logLevel    slog.Level // log level for the external processor
+	processors  []string   // list of processors to enable
+}
+
+// builtinProcessors is the map of built-in processors that can be enabled.
+var builtinProcessors = map[string]struct {
+	path    string
+	factory extproc.ProcessorFactory
+}{
+	"chatcompletions": {"/v1/chat/completions", extproc.NewChatCompletionProcessor},
+	"models":          {"/v1/models", extproc.NewModelsProcessor},
+}
+
+// parseAndValidateFlags parses and validates the flas passed to the external processor.
+func parseAndValidateFlags(args []string) (extProcFlags, error) {
+	var (
+		flags extProcFlags
+		errs  []error
+		fs    = flag.NewFlagSet("AI Gateway External Processor", flag.ContinueOnError)
+	)
+
+	fs.StringVar(&flags.configPath,
 		"configPath",
 		"",
 		"path to the configuration file. The file must be in YAML format specified in filterapi.Config type. "+
 			"The configuration file is watched for changes.",
 	)
-	extProcAddrPtr := fs.String(
+	fs.StringVar(&flags.extProcAddr,
 		"extProcAddr",
 		":1063",
 		"gRPC address for the external processor. For example, :1063 or unix:///tmp/ext_proc.sock",
 	)
 	logLevelPtr := fs.String(
 		"logLevel",
-		"info", "log level for the external processor. One of 'debug', 'info', 'warn', or 'error'.",
+		"info",
+		"log level for the external processor. One of 'debug', 'info', 'warn', or 'error'.",
+	)
+	processorsPtr := fs.String(
+		"processors",
+		"chatcompletions,models",
+		"comma-separated list of processors to enable. Available processors: chatcompletions, models",
 	)
```

Comment:
instead of exposing this in this PR, can we just use the static list for now until someone requests for exposing this as a flag? I don't think anyone wants to expose so shall we start small for now?

Reply:
Done. I've removed the flag.